yarn-project/p2p/src/services/tx_collection/tx_collection.ts

```diff
@@ -110,13 +103,17 @@ export class TxCollection {
       this.config.txCollectionReconcileIntervalMs,
     );
 
+    this.handleTxsFound = (args: Parameters<TxPoolEvents['txs-added']>[0]) => {
```

Comment:
Not a big issue, but do we need both events? It looks like we:

1. Call `foundTxs` for all txs collected by the collection sink.
2. Call `foundTxs` for all txs added to the pool and NOT collected by collection sink.

Would it work if we just listened to all txs added to the pool?